src/components/Avatar/Avatar.tsx

```diff
@@ -15,16 +16,34 @@ export const Avatar: FC<IAvatarProps> = ({
   address,
   type = 'square'
 }: IAvatarProps) => {
+  const [imageSrc, setImageSrc] = useState<string | undefined>();
+
+  useEffect(() => {
```

Comment:
Make a different component for collection cover (or extend existing one related to images) instead of this.